File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int n) {
    int s1 = 0;
    for (unsigned int i = 0; i <= n; i++) {
        s1 += i*i;
    }
    return s1;
}

unsigned int square_of_sum(unsigned int m) {
    int s2 = 0;
    for (unsigned int i = 0; i <= m; i ++) {
        s2 += i;
    }
    return s2 * s2;
}

unsigned int difference_of_squares(unsigned int a) {
    return square_of_sum(a) - sum_of_squares(a);
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    sum = 0
    for num in range(0, number+1):
        sum+= num
    return sum**2

def sum_of_squares(number):
    sum = 0
    for num in range(0, number+1):
        sum+= num ** 2
    return sum


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
